frontend/src/metabase/components/FieldValuesWidget.jsx

```diff
@@ -121,26 +135,46 @@ export class FieldValuesWidget extends Component {
     return value;
   };
 
-  search = async (value: string, cancelled: Promise<void>) => {
-    const { field, searchField, maxResults } = this.props;
+  isFieldSearchable = (field: Field) => this.searchField(field).isSearchable();
+
+  searchField = (field: Field) => {
```

Comment:
That's how I had it originally, but flow was making me check that `this.searchField(field)` was non-null all over (or add ignore comments). We only use `this.searchField` when `this.isSearchable`, so it felt a bit redundant to add those non-null checked. Though, I guess restructuring to avoid flow's wrath is just as bad.

Reply:
I only see two places where checking null would be required (`searchFieldId: this.searchField(field).id,` and  `fields.map(field => stripId(this.searchField(field).display_name)),`) 

It also seems natural to just replace `this.isFieldSearchable` with `this.searchField` here, according to the comment:

```
      // all fields have a valid search field
      fields.every(this.isFieldSearchable) &&
```

Reply:
👍 just reverted this!